File: 2199-two-furthest-houses-with-different-colors/two-furthest-houses-with-different-colors.js
/**
 * @param {number[]} colors
 * @return {number}
 */
const maxDistance = A => {
    const n = A.length;
    let left = 0, right = 0;

    for (let i = 0; i < n; i++)
        if (A[i] ^ A.at(-1)) {
            left = i;
            break;
        }

    for (let i = n - 1; i >= 0; i--)
        if (A[i] ^ A[0]) {
            right = i;
            break;
        }

    return Math.max(n - 1 - left, right);
};

File: 2199-two-furthest-houses-with-different-colors/two-furthest-houses-with-different-colors.py
class Solution:
    def maxDistance(self, A: List[int]) -> int:
        n = len(A)
        left, right = 0, n - 1

        for i in range(n):
            if A[i] ^ A[-1]:
                left = i
                break

        for i in range(n - 1, -1, -1):
            if A[i] ^ A[0]:
                right = i
                break

        return max(n - 1 - left, right)
